src/get-env-variables.js

```diff
@@ -0,0 +1,60 @@
+const exec = require('child_process').exec
+// import spawn from 'spawn-command-with-kill'
+import {assign, clone} from 'lodash'
+import managePath from 'manage-path'
+import {sync as findUpSync} from 'find-up'
+
+const NON_ERROR = 0
+
+export default getEnv
+
+function getEnv() {
+  return new Promise((resolve, reject) => {
+    getNpmEnvVariables()
+      .then(npmVars => {
+        const env = assign(clone(process.env), npmVars)
+        const alterPath = managePath(env)
+        const npmBin = findUpSync('node_modules/.bin')
+        if (npmBin) {
+          alterPath.unshift(npmBin)
+        }
+        resolve(env)
+      })
+      .catch(reject)
+  })
+}
+
+function getNpmEnvVariables() {
+  let child
+  return new Promise((resolve, reject) => {
+    let output = ''
+    child = exec('npm run env')
+
+    child.stdout.on('data', chunk => output += chunk)
+
+    // NOTE: not sure what needs to happen with error logging
+    child.stderr.on('data', data => console.log(`stderr: ${data}`))
+
+    child.on('close', code => {
+      if (code === NON_ERROR) {
+        let npmVariables = parseNpmRunEnvOutput(output)
+        resolve(npmVariables)
+      } else {
+        // NOTE: again, not sure how we want to handle errors being logged
+        reject('npm run env exited with status code:', code)
+      }
+    })
+  })
+}
+
+function parseNpmRunEnvOutput (output) {
```

Comment:
I have [a boilerplate for my projects](https://github.com/kentcdodds/generator-kcd-oss). I've often considered making `create-oss-lib`, but I've not taken the time because I find that often libs vary in needs pretty widely. May be able to get away with `create-react-lib` and `create-node-cli` or `create-js-lib`, but trying to make them all work in one package would be difficult.